lib/repl.js

```diff
@@ -571,7 +567,8 @@ REPLServer.prototype.complete = function(line, callback) {
             this.context.constructor &&
             this.context.constructor.name === 'Context') {
           var contextProto = this.context;
-          while (contextProto = Object.getPrototypeOf(contextProto)) {
+          while (
+            (contextProto = Object.getPrototypeOf(contextProto)) === true) {
```

Comment:
The common practice for those assignments is to write them like this:

```js
if ((a = 'test')) {
  // ...
}
```

So additional parentheses are a signal to a linter that a developer is intended to write this code as is.